exporter/containerimage/export.go

```diff
@@ -349,6 +349,13 @@ func (e *imageExporterInstance) pushImage(ctx context.Context, src *exporter.Sou
 }
 
 func (e *imageExporterInstance) unpackImage(ctx context.Context, img images.Image, src *exporter.Source, s session.Group) (err0 error) {
+	p := platforms.Normalize(platforms.DefaultSpec())
+	ref, ok := src.FindRef(platforms.Format(p))
+	if !ok || ref == nil {
```

Comment:
Running `docker buildx build --platform linux/amd64 -t asdf .` on `linux/arm64`:
`ok=true` and ref is not nil, so it still doesn't return early here.

Reply:
Ok, I've just gone and added an explicit check - "is the current platform in the list of platforms we built for". That should hopefully be the check that does this correctly now.